src/frontend/src/tests/lib/canisters/reward.canister.spec.ts

```diff
@@ -0,0 +1,120 @@
+import type {
+	NewVipRewardResponse,
+	_SERVICE as RewardService,
+	UserData
+} from '$declarations/rewards/rewards.did';
+import { RewardCanister } from '$lib/canisters/reward.canister';
+import type { CreateCanisterOptions } from '$lib/types/canister';
+import { mockIdentity } from '$tests/mocks/identity.mock';
+import { HttpAgent, type ActorSubclass } from '@dfinity/agent';
+import { Principal } from '@dfinity/principal';
+import { mock } from 'vitest-mock-extended';
+
+vi.mock(import('$lib/actors/agents.ic'), async (importOriginal) => {
+	const actual = await importOriginal();
+	return {
+		...actual,
+		// eslint-disable-next-line require-await
+		getAgent: async () => mock<HttpAgent>()
+	};
+});
+
+describe('reward.canister', () => {
+	const createRewardCanister = ({
+		serviceOverride
+	}: Pick<CreateCanisterOptions<RewardService>, 'serviceOverride'>): Promise<RewardCanister> =>
+		RewardCanister.create({
+			canisterId: Principal.fromText('tdxud-2yaaa-aaaad-aadiq-cai'),
+			identity: mockIdentity,
+			certifiedServiceOverride: serviceOverride,
+			serviceOverride
+		});
+	const service = mock<ActorSubclass<RewardService>>();
+	const mockResponseError = new Error('Test response error');
+
+	it('returns true if user is vip', async () => {
```

Comment:
should we update the description of the test to be more generic or should we add more tests?

When I read `returns true if user is vip` I kind of expect other test, like "user is not a vip", but since we mock the results not sure what's best.

Reply:
hmm yes. we should also provide a test for the "user is not a vip" case.